src/integration/models.ts

```diff
@@ -84,6 +84,50 @@ export const PasswordResetRequestModel = mongoose.model<PasswordResetRequest>(
   passwordResetRequestSchema,
 );
 
+// new interface for email verification
+export interface VerifyEmailRequest {
+  // verification email links generate two keys: the ID, which is stored plain, and the key, which is
+  // hashed with argon2 before being stored.
+  // good chance the rest of this shit is wrong in this implementation: 
+  // Both are provided to the user and they make a request;
+  // then we look up the request for the ID and compare hashed keys, basically exactly like a normal
+  // username/password login flow.
+  _id: string;
+  key: string;
+  user: string;
+  timestamp: Date;
+  suppressEmail?: boolean;
+}
+
+const verifyEmailRequestSchema = new mongoose.Schema<VerifyEmailRequest>({
+  _id: {
+    type: String,
+    required: true,
+  },
+  key: {
+    type: String,
+    required: true,
+  },
+  user: {
+    type: String,
+    required: true,
+    index: true, // we'll search by user to invalidate previous reset requests
+  },
+  timestamp: {
+    type: Date,
+    expires: 0,
+  },
+  suppressEmail: {
+    type: Boolean,
+    required: false,
+  },
```

Comment:
This isn't necessary for email verification requests, since there isn't really a "confirmation" of the email being verified like there is for password resets. We have this option in the password reset model because there's two scenarios where we do a password reset: 1) [explicitly creating a new account](https://github.com/swat-sccs/sauce/blob/master/src/functions/createAccount.ts#L73) and 2) [a user-requested password reset](https://github.com/swat-sccs/sauce/blob/master/src/controllers/accountController.ts#L95). In scenario 2 we send a confirmation email after the change happens (see [here](https://github.com/swat-sccs/sauce/blob/cf0766cba131d41131ac62d5b3e1cd03b67f7486/emailTemplates/passwordResetNotification.html)), but not in scenario 1.

Reply:
Wait, why was this marked as resolved? This hasn't been fixed.